# Names


In [ ]:
# Will Judge August 1st, 2020

# House Price Prediction Competition

## Abstract

First competition with fellow DSS member.
The goal is to predict the home price. 

The best technique I want to use are:

* Random Forest Regressor
* Gradient boosting
* Emsemble Learning
* Support Vector Regression

## Initialize

In [2]:
### Import libraries
import pandas as pd
import seaborn as sns
import numpy as np
from sklearn.svm import SVR

In [16]:
### Read the dataset
df = pd.read_csv('Data/train.csv')

## EDA

In [ ]:
df.shape

(1460, 81)

The train dataset contain 1460 observation and 81 columns.

# Data Prep

In [17]:
# https://pandas.pydata.org/docs/getting_started/dsintro.html
# https://dev.to/chanduthedev/how-to-display-all-rows-from-data-frame-using-pandas-dha
# https://dev.to/chanduthedev/how-to-display-all-rows-from-data-frame-using-pandas-dha
#pd.set_option('display.max_rows', 90)
#pd.DataFrame(df.columns)

df.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

In [23]:
df.columns

Index(['MSSubClass', 'MSZoning', 'LotFrontage', 'LotArea', 'Street', 'Alley',
       'LotShape', 'LandContour', 'LotConfig', 'LandSlope', 'Neighborhood',
       'Condition1', 'Condition2', 'BldgType', 'HouseStyle', 'OverallQual',
       'OverallCond', 'YearBuilt', 'YearRemodAdd', 'RoofStyle', 'RoofMatl',
       'Exterior1st', 'Exterior2nd', 'MasVnrType', 'MasVnrArea', 'ExterQual',
       'ExterCond', 'Foundation', 'BsmtQual', 'BsmtCond', 'BsmtExposure',
       'BsmtFinType1', 'BsmtFinSF1', 'BsmtFinType2', 'BsmtFinSF2', 'BsmtUnfSF',
       'TotalBsmtSF', 'HeatingQC', 'CentralAir', 'Electrical', '1stFlrSF',
       '2ndFlrSF', 'LowQualFinSF', 'GrLivArea', 'BsmtFullBath', 'BsmtHalfBath',
       'FullBath', 'HalfBath', 'BedroomAbvGr', 'KitchenAbvGr', 'KitchenQual',
       'TotRmsAbvGrd', 'Fireplaces', 'FireplaceQu', 'GarageType',
       'GarageYrBlt', 'GarageFinish', 'GarageCars', 'GarageQual', 'GarageCond',
       'PavedDrive', 'WoodDeckSF', 'OpenPorchSF', 'EnclosedPorch', '3SsnPorch',
   

In [21]:
# UnImportant Columns?
# The ID Number is just for order purposes - has no berring on house price
# Utilities - does it really matter where the utilities come from?
# A majority of houses have heating
# What is functional?
# GarageArea is directly related to number of cars
# SaleType might not matter
unimportant = ['Id', 'Utilities', 'Heating', 'Functional', 'GarageArea', 'SaleType']

for remove_item in unimportant:
    df.pop(remove_item)
    
df.head()

MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0          60       RL         65.0     8450   Pave   NaN      Reg   
1          20       RL         80.0     9600   Pave   NaN      Reg   
2          60       RL         68.0    11250   Pave   NaN      IR1   
3          70       RL         60.0     9550   Pave   NaN      IR1   
4          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour LotConfig LandSlope  ... ScreenPorch PoolArea PoolQC Fence  \
0         Lvl    Inside       Gtl  ...           0        0    NaN   NaN   
1         Lvl       FR2       Gtl  ...           0        0    NaN   NaN   
2         Lvl    Inside       Gtl  ...           0        0    NaN   NaN   
3         Lvl    Corner       Gtl  ...           0        0    NaN   NaN   
4         Lvl       FR2       Gtl  ...           0        0    NaN   NaN   

  MiscFeature  MiscVal  MoSold  YrSold  SaleCondition SalePrice  
0         NaN        0       2    2008         Normal    208500  
1         NaN        0       5    2007         Normal    181500  
2         NaN        0       9    2008         Normal    223500  
3         NaN        0       2    2006        Abnorml    140000  
4         NaN        0      12    2008         Normal    250000  

[5 rows x 75 columns]

In [ ]:
## Examine the target variable: Sales Price
# https://www.kaggle.com/pmarcelino/comprehensive-data-exploration-with-python

## References

### Python

### Panda

* https://pandas.pydata.org/docs/getting_started/index.html
* https://pandas.pydata.org/docs/getting_started/comparison/comparison_with_r.html#compare-with-r

### Machine Learning

### Kaggles notebooks

* https://www.kaggle.com/pmarcelino/comprehensive-data-exploration-with-python

# Random Forest Regressor Training

# Gradient Boosting Training

# Emsemble Learning Training

# Support Vector Regression Training